app/models/manager_refresh/inventory_object_lazy.rb

```diff
@@ -28,24 +28,24 @@ def load
     def dependency?
       # If key is not set, InventoryObjectLazy is a dependency, cause it points to the record itself. Otherwise
       # InventoryObjectLazy is a dependency only if it points to an attribute which is a dependency or a relation.
-      !!(!key || transitive_dependency?) && !inventory_collection.saved?
+      !key || transitive_dependency?
     end
 
     def transitive_dependency?
       # If the dependency is inventory_collection.lazy_find(:ems_ref, :key => :stack)
       # and a :stack is a relation to another object, in the InventoryObject object,
       # then this relation is considered transitive.
-      !!(key && (inventory_collection.dependency_attributes.keys.include?(key) ||
-        association?(key)))
+      key && association?(key)
     end
 
     def association?(key)
       # TODO(lsmola) remove this if there will be better dependency scan, probably with transitive dependencies filled
-      # in a second pass
+      # in a second pass, then we can get rid of this hardcoded symbols. Right now we are not able to introspect these.
       return true if [:parent, :genelogy_parent].include?(key)
 
-      # Is the key an association on inventory_collection_scope model class?
-      !inventory_collection.association_to_foreign_key_mapping[key].nil?
+      # Is the key a dependency or an association on inventory_collection model class?
```

Comment:
fyi: this is hard to read. unknown if this is a developer question, or declaring the action.

maybe:

```
Return true if the key is a dependency or association on inventory_collection model class.
```

Also, this does not belong here but rather in in the method description